core/deployment/src/main/java/io/quarkus/deployment/jbang/JBangAugmentorImpl.java

```diff
@@ -112,6 +114,7 @@ public void accept(BuildChainBuilder builder) {
             List javaargs = new ArrayList<String>();
             javaargs.add("-Djava.util.logging.manager=org.jboss.logmanager.LogManager");
             resultMap.put("java-args", javaargs);
+            resultMap.put("main-class", buildResult.consume(MainClassBuildItem.class).getClassName());
```

Comment:
this one i assume will either be a generated one or the user provided one, correct?

Reply:
yes